src/proxyvisor/proxyvisor-v1.coffee

```diff
@@ -0,0 +1,182 @@
+Promise = require 'bluebird'
+_ = require 'lodash'
+express = require 'express'
+fs = Promise.promisifyAll require 'fs'
+# { request } = require '../lib/request'
+# constants = require '../lib/constants'
+bodyParser = require 'body-parser'
+url = require 'url'
+{
+	isDefined,
+	parseDeviceFields,
+	validStringOrUndefined,
+	validObjectOrUndefined,
+	# tarDirectory,
+	# tarFilename,
+	tarPath,
+	getTarArchive,
+	# cleanupTars,
+	# formatTargetAsState,
+	# formatCurrentAsState,
+} = require './lib'
+
+# Does this need to be a class? I tried without and routing worked but couldn't access any of the other objects e.g. @db
+module.exports = class ProxyvisorRouterV1
+	constructor: (@proxyvisor) ->
+		console.log('constructor')
+		{ @config, @logger, @db, @docker, @reportCurrentState } = @proxyvisor
+		@router = express.Router()
+		@router.use(bodyParser.urlencoded(extended: true))
+		@router.use(bodyParser.json())
+
+	@router.get '/test', (req, res) ->
```

Comment:
All of these need one more indentation level, as they're actually part of the constructor.